### PR TITLE
Typography Changes

### PR DESCRIPTION
**Pull request by prestonbourne**
[After Merge](https://swingset-example-git-prestypography-styling-hashicorp.vercel.app/swingset)
[Current](https://swingset-example.vercel.app/swingset)

## Style Tweaks
Small PR that begins pushing towards the style system laid out by Ben

### Changes
- Rename custom tailwind styles
- Create text components `Body` and `Heading` 

### Next Steps
- I started to also adjust the font styles on the left nav but I figured it was better it change everything related to that component in it's own PR
- Discovered this small [issue](https://github.com/hashicorp/swingset/issues/128)

### CODE REVIEW

File: packages/swingset-theme-hashicorp/src/components/side-nav/category.tsx [hunk 1]
```diff
@@ -21,8 +22,8 @@ function Category({
 //Swap this out for already existing heading && Enquire about semantics, https://helios.hashicorp.design/components/application-state uses <div>
 function CategoryHeading({ children }: { children: string }) {
   return (
-    <div className="ss-uppercase ss-text-xs ss-font-semibold ss-leading-6 ss-text-foreground-faint ss-border-b ss-border-faint ss-pb-2">
-      {children.toUpperCase()}
+    <div className="ss-capitalize ss-text-xs ss-font-semibold ss-leading-6 ss-text-primary ss-border-b ss-border-faint ss-pb-2">
+      {children}
```
Comment (prestonbourne) [inline packages/swingset-theme-hashicorp/src/components/side-nav/category.tsx:26]: We currently capitalize the entire word but Ben's design uses standard capitalization

File: packages/swingset-theme-hashicorp/tailwind.config.cjs [hunk 2]
```diff
@@ -7,11 +9,11 @@ module.exports = {
...
     extend: {
-      colors: {},
       textColor: {
-        'foreground-primary': '#3b3d45',
-        'foreground-faint': '#656a76',
-        'foreground-action': '#1060ff',
+        dark: colors.slate[700],
+        primary: colors.slate[600],
+        faint: colors.gray[500],
+        action: '#1060ff',
```
Comment (prestonbourne) [inline packages/swingset-theme-hashicorp/tailwind.config.cjs:16]: When we pass a custom class into the `textColor` sub-object. Tailwind exposes the style as `text-{className}`.

So currently, the class name is `ss-text-foreground-primary`. Foreground and text is a redundancy.

File: packages/swingset-theme-hashicorp/src/components/side-nav/category.tsx [hunk 3]
```diff
@@ -9,10 +9,11 @@ function Category({
   title: string
   items: NavigationNode[]
 }) {
+  console.log(title)
```
Comment (dstaley) [inline packages/swingset-theme-hashicorp/src/components/side-nav/category.tsx:12]: leftover console.log

File: packages/swingset-theme-hashicorp/src/components/text/body.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { cva, VariantProps, cx } from 'class-variance-authority'
+import { type HTMLAttributes } from 'react'
+
+type HeadingProps = HTMLAttributes<HTMLHeadingElement>
```
Comment (dstaley) [inline packages/swingset-theme-hashicorp/src/components/text/body.tsx:4]: `HTMLHeadingElement` is for `h1, h2, h3, etc`, but these props are being passed to a `HTMLParagraphElement`.

File: packages/swingset-theme-hashicorp/src/components/text/heading.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      as: 'h3',
+    },
+  }
+)
+
+type HeadingProps = HTMLAttributes<HTMLHeadingElement> &
+  VariantProps<typeof styles>
+
+export function Heading({ as, children, className, ...props }: HeadingProps) {
+  const Element = as ?? 'h3'
```
Comment (dstaley) [inline packages/swingset-theme-hashicorp/src/components/text/heading.tsx:27]: Why is the default `h3`?

**Review by dstaley — CHANGES_REQUESTED** [review]
Looks good! Just a few small cleanup things.

**Pull request merged**